src/tools/perf/perftest_daemon.c

```diff
@@ -96,6 +107,288 @@ ucp_perf_daemon_server_conn_handle_cb(ucp_conn_request_h conn_request,
     }
 }
 
+static void
+ucp_perf_daemon_set_am_recv_handler(ucp_perf_daemon_context_t *ctx,
+                                    ucp_perf_daemon_am_id_t am_id,
+                                    ucp_am_recv_callback_t cb)
+{
+    ucp_am_handler_param_t param = {};
+    ucs_status_t status;
+
+    param.field_mask = UCP_AM_HANDLER_PARAM_FIELD_ID    |
+                       UCP_AM_HANDLER_PARAM_FIELD_CB    |
+                       UCP_AM_HANDLER_PARAM_FIELD_FLAGS |
+                       UCP_AM_HANDLER_PARAM_FIELD_ARG;
+    param.id         = am_id;
+    param.cb         = cb;
+    param.flags      = UCP_AM_FLAG_WHOLE_MSG;
+    param.arg        = ctx;
+    status           = ucp_worker_set_am_recv_handler(ctx->worker, &param);
+
+    ucs_assertv_always(status == UCS_OK, "status=%s", ucs_status_string(status));
```

Comment:
probably need better error handling flow than fatal assertion

Reply:
can it be error message without fatal? and propagate the error to main loop, event without proper handling?
fatal error can create core dump, backtrace, etc, and not sure this is the right way here

Reply:
ok, will be fixed